examples/nestjs/consumer/test/app.spec.ts

```diff
@@ -0,0 +1,267 @@
+
+          expect(suggestedMates).toHaveProperty('suggestions');
+
+          const { suggestions } = suggestedMates;
+
+          expect(suggestions).toHaveLength(MIN_ANIMALS);
+          expect(suggestions[0].score).toBe(94);
+        });
+      });
+    });
+  });
+
+  describe('when a call to the Animal Service is made to retreive a single animal by ID', () => {
+    describe('and there is an animal in the DB with ID 1', () => {
+      beforeAll(() =>
+        provider.addInteraction({
+          state: 'Has an animal with ID 1',
+          uponReceiving: 'a request for an animal with ID 1',
+          withRequest: {
+            method: HTTPMethod.GET,
+            path: term({ generate: '/animals/1', matcher: '/animals/[0-9]+' }),
+            headers: { Authorization: 'Bearer token' },
+          },
+          willRespondWith: {
+            status: HttpStatus.OK,
+            headers: {
+              'Content-Type': 'application/json; charset=utf-8',
+            },
+            body: animalBodyExpectation,
+          },
+        }),
+      );
+
+      it('returns the animal', async () => {
+        const suggestedMates = await animalsService.getAnimalById(11);
+
+        expect(suggestedMates).toHaveProperty('id', 1);
+      });
+    });
+
+    describe('and there no animals in the database', () => {
+      beforeAll(() =>
+        provider.addInteraction({
+          state: 'Has no animals',
+          uponReceiving: 'a request for an animal with ID 100',
+          withRequest: {
+            method: HTTPMethod.GET,
+            path: '/animals/100',
+            headers: { Authorization: 'Bearer token' },
+          },
+          willRespondWith: {
+            status: HttpStatus.NOT_FOUND,
+          },
+        }),
+      );
+
+      it('returns a 404', async () => {
+        // uncomment below to test a failed verify
+        // const suggestedMates = await animalsService.getAnimalById(123)
+        const suggestedMates = animalsService.getAnimalById(100);
+
+        await expect(suggestedMates).rejects.toThrowError();
+      });
+    });
+  });
+
+  describe('when a call to the Animal Service is made to create a new mate', () => {
+    beforeAll(() =>
+      provider.addInteraction({
+        state: undefined,
+        uponReceiving: 'a request to create a new mate',
+        withRequest: {
+          method: HTTPMethod.POST,
+          path: '/animals',
+          body: like(suitor),
+          headers: {
+            'Content-Type': 'application/json; charset=utf-8',
+          },
+        },
+        willRespondWith: {
+          status: HttpStatus.OK,
+          headers: {
+            'Content-Type': 'application/json; charset=utf-8',
+          },
+          body: like(suitor),
+        },
+      }),
+    );
+
+    it('creates a new mate', async () => {
+      return expect(
+        animalsService.createMateForDates(suitor),
+      ).resolves.not.toThrow();
```

Comment:
I think it would be better to check that you unmarshalled the response correctly here - something like `.resolves.toBe(suitor)`

Also, a nitpick- there's a bit of inconsistency in the style of testing promises here - personally, I reckon the tests would read better if all assertions were in one style (also this one is marked `async` but has no await).